erpnext/public/js/controllers/buying.js

```diff
@@ -200,6 +192,41 @@ erpnext.buying.BuyingController = erpnext.TransactionController.extend({
 		this.calculate_accepted_qty(doc, cdt, cdn)
 	},
 
+	set_item_query: function() {
+		var me = this;
+		this.frm.fields_dict.items.grid.set_custom_query = function () {
+			me.frm.set_query("item_code", "items", function (doc) {
+				return me.get_item_query(doc);
+			});
+		}
+
+		this.frm.set_query("item_code", "items", function (doc) {
+			return me.get_item_query(doc);
+		});
+	},
+
+	get_item_query: function (doc) {
+		if (doc.supplier && doc.filter_items_by_supplier) {
```

Comment:
Have we tested if this breaks in doctypes where this field doesn't exist?

Reply:
We tested this in Supplier Quotation and MREQ, and it doesn't break since all missing fields default to `undefined` in JS